src/helpViewer/helpProvider.ts

```diff
@@ -34,9 +34,10 @@ export class HelpProvider {
         this.port = this.launchRHelpServer(); // is a promise for now!
     }
 
-    public refresh(): void {
+    public async refresh(): Promise<void> {
         kill(this.cp.pid); // more reliable than cp.kill (?)
```

Comment:
I'm wondering if it could kill the wrong process?

Say we start R help server (pid=12345) and `this.cp.pid` remembers it. And we manually kill the process. Then another user process starts with pid=12345. Now if I click refresh cache & restart help server, it will kill the wrong process as the pid is used by another process.

When the process is killed externally, the only difference here seems to the value of `cp.signalCode` is `'SIGTERM'`.

Reply:
I added a check to see if the cp process has exited (for whatever reason), before killing it.